### DISCUSSION TITLE
pagination / special handling of page 1

### DISCUSSION THREAD

**Discussion by veesahni** [withastro/roadmap#385] [Proposal]
[Astro has built in pagination support](https://docs.astro.build/en/core-concepts/routing/#the-page-prop)

However, I don't want my first page to be `/blog/1`, I rather it be just `/blog` 

I'm not seeing any nice/clean way to do.

One option:
* Create a component `BlogPage(page)` that is used in `blog/[page].astro` and also in `blog/index.astro`
* Now I have to manually create a fake `page `object to send to to `BlogPage()` in `blog/index.astro` ... would be nice to be able to do something like `page = Astro.paginate(data, pageCount).page(1)`
* This also has problem that /blog/1 does exist, so i'll need to remove it from dist after the build step.... and page 2's back button will have to have page number removed as part of rendering.

Thoughts? Did I miss something in the docs?

**Comment by delucis** [comment]
Not sure if it helps here, but with rest parameters like `/blog/[...page].astro`, you can return `params: { page: undefined }` to create `/blog/index.html`, docs here: <https://docs.astro.build/en/core-concepts/routing/#rest-parameters> and we’ve been discussing some improvements to those docs in this issue: https://github.com/withastro/docs/issues/2013

**Comment by veesahni** [comment]
@delucis I did some tests (on astro 1.6.7) and it seems like pagination with rest params is already working the way I described out of the box.

To be clear (for anyone who stumbles on this thread), take the pagination example from the docs, change filename [page].astro to [...page].astro and the built in paginate() function outputs undefined for page 1 - thus /blog and blog/2, with next/prev all functional.

**Comment by eldair** [comment]
This is still relevant (@veesahni comment) because I cannot find it in docs.

**Reply by florian-lefebvre** [reply]
It's documented at https://docs.astro.build/en/reference/routing-reference/#paginate:

> /posts/[...page].astro would generate the URLs /posts, /posts/2, /posts/3, etc.

**Reply by eldair** [reply]
Thanks a lot! I might try to improve pagination in guides part of the docs to include this

**Reply by florian-lefebvre** [reply]
You're welcome! Please open a discussion on the docs first: https://github.com/withastro/docs/discussions
